hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterRpcServer.java

```diff
@@ -1765,6 +1765,9 @@ protected List<RemoteLocation> getLocationsForPath(String path,
           locs.add(loc);
         }
       }
+      if (locs.isEmpty()) {
+        throw new NoLocationException(path, this.subclusterResolver.getClass().getSimpleName());
```

Comment:
Cleaner to pass this.subclusterResolver.getClass() and do the getSimpleName() in the exception.

Reply:
fixed